Make OpenWrt more SPDX compliant

aparcar opened the pull request:
The license folder is a core part of OpenWrt and all GPL-2.0 licensed.
Adding a SPDX license tag allows machines to check licenses.

The `(C)` is optional based on the GPS 2.0 text, so why waste 3 bytes on
every text file. Update copyright years to 2020, this should be done in
a recurring manner, e.g. yearly.

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar committed:
treewide: add SPDX license tags

The license folder is a core part of OpenWrt and all GPL-2.0 licensed.
Adding a SPDX license tag allows machines to check licenses.

The `(C)` is optional based on the GPS 2.0 text, so why waste 3 bytes on
every text file. Update copyright years to 2020, this should be done in
a recurring manner, e.g. yearly.

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar committed:
build: add missing licenses to LICENSES dir

OpenWrt ships code not only licensed via GPL-2.0-{only,or-later}.
Reflect that by adding all missing licenses, just like the kernel does.

Signed-off-by: Paul Spooren <mail@aparcar.org>

aparcar commented:
# SUMMARY

* Bad licenses:
* Deprecated licenses:
* Licenses without file extension:
* Missing licenses: GPL-2.0, GPL-2.0+
* Unused licenses:
* Used licenses: BSD-2-Clause, BSD-3-Clause, GPL-1.0-or-later, GPL-2.0-only, GPL-2.0-or-later, ISC, Linux-syscall-note, MIT
* Read errors: 1
* Files with copyright information: 2082 / 9920
* Files with license information: 973 / 9920

adschm commented on include/autotools.mk:
```diff
@@ -1,9 +1,6 @@
-# This is free software, licensed under the GNU General Public License v2.
-# See /LICENSE for more information.
-#
+# SPDX-License-Identifier: GPL-2.0-or-later
```
I think the SPDX license should be in the first line, and the copyright may come afterwards. That's also what the kernel recommendations say:

[...]
The SPDX license identifier in kernel files shall be added at the first possible line in a file which can contain a comment. For the majority or files this is the first line, except for scripts which require the ‘#!PATH_TO_INTERPRETER’ in the first line. For those scripts the SPDX identifier goes into the second line.
[...]

I'm sorry for creating additional work by this ...

aparcar replied:
Yea I remember that too, wonder why the `reuse` tools doesn't follow this idea. I'll patch it

adschm commented on include/depends.mk:
```diff
@@ -1,15 +1,13 @@
+# 	1: directories/files
+# 	2: directory dependency
+# 	3: tempfile for file listings
+# 	4: find options
```
What's that?

aparcar replied:
Fixed

adschm commented on LICENSE:
```diff
@@ -1,339 +0,0 @@
-                    GNU GENERAL PUBLIC LICENSE
```
Like for README, please check whether removing this file will break anything ...

adschm commented:
I just pushed the scripts string update and did the target update myself (without touching patches, though).

aparcar committed:
scripts: fix checkpatch.pl for changed license dir

As multiple LICENSES are shipped and no longer just LICENSE, modify the
OpenWrt tree detecation in checkpatch.pl.

Signed-off-by: Paul Spooren <mail@aparcar.org>

adschm commented on include/autotools.mk:
```diff
@@ -1,9 +1,6 @@
 #
-# Copyright (C) 2007-2012 OpenWrt.org
-#
-# This is free software, licensed under the GNU General Public License v2.
```
Here (and below) you are converting GPL-2.0 to GPL-2.0-or-later?

adschm commented:
please don't use spdx as prefix, but refer to the folders/code hierarchy as for other patches ...

adschm commented on target/linux/ath79/patches-4.19/0028-MIPS-ath79-drop-machfiles.patch:
```diff
@@ -42,7 +42,7 @@ Signed-off-by: John Crispin <john@phrozen.org>
 +++ b/arch/mips/ath79/Kconfig
 @@ -1,79 +1,6 @@
- # SPDX-License-Identifier: GPL-2.0
+ # SPDX-License-Identifier: GPL-2.0-only
```
This line is from kernel, we don't add/remove it, but just use it to determine the position of the changes. Lines like this must not be changed.

adschm replied:
And personally I'm not sure whether we should touch the patches at all.

aparcar replied:
I removed the target patch for now.

aparcar commented:
> please don't use spdx as prefix, but refer to the folders/code hierarchy as for other patches ...

Done

neheb commented:
Relevant: https://github.com/systemd/systemd/commit/810adae9e9e0b028bd1d210918d0380e15c5595a

aparcar commented:
Thanks! However I was planing to update to a more modern Standard https://emojipedia.org/copyright/

aparcar commented:
> This important patch is very important.

;)

adschm commented:
I've personally never understood why people use "Copyright (C)" or similar as it's redundant to me. One also don't say it's 30 kilograms (kg) ...
However, as it's essentially completely irrelevant, I'd only _change_ it if the new variant is superior for some objective reason, or we really follow a common standard afterwards.

aparcar commented:
GPL says it's optional, let's make it the standard.

f00b4r0 commented:
IANAL but ISTR that the "(C)" is actually required in some legislations. i.e. that's not a license thing, that's a law one, and I'd thread very carefully there. IMHO the size argument cuts both ways.

aparcar commented:
Well the GPL text says it's optional, but I don't really care. @adschm preferences?

adschm commented:
As stated before, I wouldn't do a purely "cosmetic" change without a good reason.

adschm commented:
Looking at this again, I'm still not convinced we should add all those license headers where they have been missing so far.

If we can deduce the type of the files' licenses from our "project license", I don't see a point of adding them to the files, as we know the license already. And if we cannot deduce anything, we cannot add anything.

So, I'd really just limit this to a mere replacement of long text licenses to license identifiers/headers, and that's it.

aparcar commented:
@adschm I updated the files so we use spdx where there was already a license and keep the unlicensed files as before. I'll write the SFC a email asking what to do in such cases.

aparcar commented:
@adschm I wrote the SFC an email and added more files from `config/`, please merge this and once SFC has shared an opinion I'll do some follow ups

bkuhn commented:
I sent an email about this to @aparcar last week.  This issue is complicated and we should probably have an email thread about it that will be easier to discuss these complex issues here.  @aparcar, feel free to add to the Cc list anyone you think should be included in that thread (e.g., @olek2).

CodeFetch commented on LICENSES/BSD-2-Clause.txt:
```diff
@@ -0,0 +1,22 @@
+Copyright (c) <year> <owner>. All rights reserved.
```
Shouldn't there be a year given?

aparcar replied:
From my understanding that's the license text and year plus text come from the individual files.

aparcar commented:
@adschm please merge the second commit

adschm commented:
I have reshaped this a little and pasted it here:

https://patchwork.ozlabs.org/project/openwrt/list/?series=229295
